proxy/grpcproxy/watcher_groups.go

```diff
@@ -54,9 +55,10 @@ func (wgs *watchergroups) addWatcher(rid receiverID, w watcher) {
 	watchg.add(rid, w)
 	go watchg.run()
 	groups[w.wr] = watchg
+	wgs.idToGroup[rid] = watchg
 }
 
-func (wgs *watchergroups) removeWatcher(rid receiverID) bool {
+func (wgs *watchergroups) removeWatcher(rid receiverID) (int64, bool) {
```

Comment:
after a second thought while changing the code, i feel we should keep the bool. Or we will have rev != -1 all over the place. and it is really clear why we need to compare rev to -1.